app/uk/gov/hmrc/helptosavefrontend/forms/BankDetailsValidation.scala

```diff
@@ -16,23 +16,27 @@
 
 package uk.gov.hmrc.helptosavefrontend.forms
 
-import cats.data.Validated.{Invalid, Valid}
+import java.util.regex.Matcher
+
+import cats.data.Validated.Valid
 import cats.instances.int._
 import cats.instances.string._
 import cats.syntax.either._
 import cats.syntax.eq._
 import com.google.inject.Inject
-import play.api.data.{Form, FormError}
 import play.api.data.Forms.{optional, text}
 import play.api.data.format.Formatter
+import play.api.data.{Form, FormError}
 import uk.gov.hmrc.helptosavefrontend.config.FrontendAppConfig
-import uk.gov.hmrc.helptosavefrontend.util.Validation.{ValidOrErrorStrings, invalid, validatedFromBoolean}
 import uk.gov.hmrc.helptosavefrontend.forms.BankDetailsValidation.{ErrorMessages, StringOps}
+import uk.gov.hmrc.helptosavefrontend.util.Validation.{ValidOrErrorStrings, invalid, validatedFromBoolean}
 
 class BankDetailsValidation @Inject() (configuration: FrontendAppConfig) {
 
   import configuration.BankDetailsConfig._
 
+  private val rollNoRegex: String ⇒ Matcher = (s"^([0-9a-zA-Z\\\\/\\\\.-]{$rollNumberMinLength,$rollNumberMaxLength})" + "$").r.pattern.matcher _
```

Comment:
How sure are you about this regex? especially the ```\\\\/\\\\``` bit?

Reply:
my Intellij automatically adding the extra `\\` , let me check the atos spec

Reply:
ok

Reply:
i checked in the https://regex101.com/ and i have a feeling that ^([0-9a-zA-Z\/.-]{4, 18})$ is the correct one to use
but the spec has \\/\\